src/exec/api-threads.k

```diff
@@ -88,9 +87,13 @@ K performance evaluation with the rule below active and commented:
 
   Conclusion: the rule below have no impact on performance.
 */
+//todo mapItemByValue/solution
+//rule [monitor-completely-released]:
+//    <holds>... (OL:Int |-> 0 => .Map) ...</holds>
+//    <busy>... (SetItem(OL) => .Set) ...</busy>
 rule [monitor-completely-released]:
-    <holds>... (OL:Int |-> 0 => .) ...</holds>
-    <busy>... (SetItem(OL) => .) ...</busy>
+    <holds> M:Map => removeMapItemsByValue(M,0) </holds>
+    <busy> S:Set => S -Set keys(M -Map removeMapItemsByValue(M,0)) </busy>
```

Comment:
Does ```<busy>``` contain the keys of the non-zero entries in ```<holds>```? If that is true, then you could set ```<busy>``` to be ```keys(removeMapItemsByValue(M,0))```.